internal/webpush/realsender.go

```
@@ -0,0 +1,315 @@
+	case gtsmodel.NotificationPendingReblog:
+		pushNotification.Title = fmt.Sprintf("%s boosted your post, which requires your approval", displayNameOrAcct)
+	case gtsmodel.NotificationAdminReport:
+		pushNotification.Title = fmt.Sprintf("%s submitted a report", displayNameOrAcct)
+	case gtsmodel.NotificationUpdate:
+		pushNotification.Title = fmt.Sprintf("%s updated their post", displayNameOrAcct)
+	default:
+		log.Warnf(ctx, "Unknown notification type: %d", notification.NotificationType)
+		pushNotification.Title = fmt.Sprintf(
+			"%s did something (unknown notification type %d)",
+			displayNameOrAcct,
+			notification.NotificationType,
+		)
+	}
+
+	// Set the notification body.
+	if apiNotification.Status != nil {
+		if apiNotification.Status.SpoilerText != "" {
+			pushNotification.Body = apiNotification.Status.SpoilerText
+		} else {
+			pushNotification.Body = text.SanitizeToPlaintext(apiNotification.Status.Content)
+		}
+	} else {
+		pushNotification.Body = text.SanitizeToPlaintext(apiNotification.Account.Note)
+	}
+	pushNotification.Body = firstNBytesTrimSpace(pushNotification.Body, bodyMaxLen)
+
+	// Encode the push notification as JSON.
+	pushNotificationBytes, err := json.Marshal(pushNotification)
+	if err != nil {
+		return gtserror.Newf("error encoding Web Push notification: %w", err)
+	}
+
+	// Send push notification.
+	resp, err := webpushgo.SendNotificationWithContext(
+		ctx,
+		pushNotificationBytes,
+		&webpushgo.Subscription{
+			Endpoint: subscription.Endpoint,
+			Keys: webpushgo.Keys{
+				Auth:   subscription.Auth,
+				P256dh: subscription.P256dh,
+			},
+		},
+		&webpushgo.Options{
+			HTTPClient:      r.httpClient,
+			Subscriber:      vapidSubjectEmail,
+			VAPIDPublicKey:  vapidKeyPair.Public,
+			VAPIDPrivateKey: vapidKeyPair.Private,
+			TTL:             int(TTL.Seconds()),
+		},
+	)
+	if err != nil {
+		return gtserror.Newf("error sending Web Push notification: %w", err)
+	}
+	defer func() {
```

You can just `defer resp.Body.Close()`

---

Sure, I guess it's idiomatic enough that we don't need to explicitly ignore the error.